### PR TITLE
Cleanup III

### PR DESCRIPTION
**Pull request by RickiJay-WMDE**
- Renamed files for consistent kebab-case
- Updated `lint.sh` to lint Markdown, Python, and YML files across the repo
- Separated linting steps: Javascript/Typescript/JSON, Markdown, YML
- Renamed spec files for accuracy and clarity
- `expect().toHaveTextContaining()` is deprecated in the next `@wdio` version; updated to use RegExp in their place

### CODE REVIEW

**Review by rti — CHANGES_REQUESTED** [review]
🚀🚀🚀

File: test/.eslintrc.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
 		"wikimedia/language/es2019",
 		"@wmde/wikimedia-typescript"
 	],
-	"ignorePatterns": [ "**/results/result.json" ],
+	"ignorePatterns": [ "**/results/result.json", "!.github", "!example" ],
```
[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply test/.eslintrc.json:8]: Exactly, yes

File: .github/reporter/report.js [hunk 2]
```diff
@@ -1,49 +1,53 @@
...
+const core = require( '@actions/core' );
+const { issueCommand } = require( '@actions/core/lib/command' );
 
-const suiteName = process.argv[2];
+const suiteName = process.argv[ 2 ];
 
-if( suiteName ) {
-    return;
+if ( suiteName ) {
+	return;
```
[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply .github/reporter/report.js:11]: It's a tab character - and I'm pretty sure you can customize how many spaces it appears as in github

File: lint.sh [hunk 3]
```diff
@@ -13,16 +13,60 @@ done
...
+#   -v "$(pwd)/.github":/usr/src/test/github \
+#   -v "$(pwd)/docs":/usr/src/test/docs \
+#   -v "$(pwd)/example":/usr/src/test/example \
+#   test-runner -c "$NPM_LINT_COMMAND"
+
+# ℹ️ Linting Javascript test/**/*.cjs,js,json,mjs,ts
+if $SHOULD_FIX
+then
+  $RUN_TEST_RUNNER_CMD "
+    npx prettier \"**/*.{cjs,js,json,mjs,ts}\" --write \
```
[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply lint.sh:30]: The argument @lorenjohnson had for this was any Javascript elsewhere would be in its own Node project - which is true, the only other JS present is in `.github/reporter` - and therefore those projects should have their own linting. That one currently doesn't, admittedly.

File: lint.sh [hunk 4]
```diff
@@ -13,16 +13,60 @@ done
...
+      --log-level error
+  "
+fi
+$RUN_TEST_RUNNER_CMD "
+  npx eslint . --ext .cjs,.js,.json,.mjs,.ts $ESLINT_FLAGS
```
[the remark this answers is not in the document]
Comment (RickiJay-WMDE) [reply lint.sh:35]: See previous

File: lint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (rti) [reply lint.sh:30]: This actually brings me back to "linting should be handled outside of test on the root of our project". Then it makes sense again I think.

**Pull request merged**